src/common/components/repository-row/index.js

```diff
@@ -354,31 +357,29 @@ class RepositoryRow extends Component {
 
 RepositoryRow.propTypes = {
   snap: PropTypes.shape({
-    resource_type_link: PropTypes.string,
     git_repository_url: PropTypes.string,
-    self_link: PropTypes.string,
     store_name: PropTypes.string,
     snapcraft_data: PropTypes.object
   }),
   latestBuild: PropTypes.shape({
     buildId: PropTypes.string,
-    status: PropTypes.string,
+    buildLogUrl: PropTypes.string,
+    colour: PropTypes.string,
+    dateStarted: PropTypes.string,
     statusMessage: PropTypes.string
   }),
   fullName: PropTypes.string,
   authStore: PropTypes.shape({
     authenticated: PropTypes.bool,
-    hasDischarge: PropTypes.bool,
-    isFetching: PropTypes.bool,
-    signedAgreement: PropTypes.bool,
     userName: PropTypes.string
   }),
   registerNameStatus: PropTypes.shape({
-    isFetching: PropTypes.bool,
-    success: PropTypes.bool,
-    error: PropTypes.object
+    success: PropTypes.bool
   }),
   dispatch: PropTypes.func.isRequired
 };
 
+// FIXME:
+// `connect` shouldn't be used just to add dispatch to props
+// this makes it much harder to test it as it get wrapped
 export default connect()(RepositoryRow);
```

Comment:
I think that's an antipattern we should avoid.

`RepositoryRow` gets all it's data in props from parent component, so in theory could be a pure presentational component. This `connect` here doesn't really bind it to data, it's only used to pass `dispatch`, so it can call various redux actions (which it does quite a lot, as there is lot's of functionality there).

I don't want to change it now (that's why I just added a FIXME), but I think it should be either pure react component that will not get dispatch from connect (so we either pass dispatch or different action handlers from parent component) or we make it into fully connected component that reads its data from the store.

But that's probably something that can be part of component structure discussion we have in plans.